src/utils.ts

```diff
@@ -100,3 +109,41 @@ export async function sendData (url :string, ciTelemetryData: CITelemetryData)
         }
       }
 }
+
+async function getApiKey (metaData: MetaData) : Promise<string | null>
+{
+    const apiKey : string = core.getInput("api_key");
+    if (apiKey) {
+      logger.debug(`ApiKey: ${apiKey}`);
+      return apiKey;
+    } else {
+      logger.debug(`ApiKey is not defined! Requesting on demand ApiKey`);
+      const onDemandApiKey = await getOnDemandApiKey( metaData);
+      return (onDemandApiKey != null) ? onDemandApiKey : null
+    }
+}
+
+async function getOnDemandApiKey (metaData: MetaData) : Promise<string | null>
+{
+    logger.debug(`Getting on demand api key from: ${ON_DEMAND_API_KEY_ENDPOINT}`);
+    try {
+        const { data } =  await axios.post(
+          ON_DEMAND_API_KEY_ENDPOINT,
+          metaData,
+          {
+            headers: {
+              'Content-type': 'application/json; charset=utf-8'
+            },
+          },
+        );
+        logger.debug(`Data: ${data}`)
+        return data;
```

Comment:
Seems that we return API key as string in the response. But I think it is better to return in a json like this:
```json
{
    apiKey: $onDemandAPIKey
}
```

So, we can add additional info in the future like expire time, etc ...

Reply:
backend api will returns data in this format